setup.py

```diff
@@ -10,6 +10,7 @@
     "awkward": ["awkward>=1.2.0"],
     "test": [
         "pytest>=4.6",
+        "pytest-cov==3.0.0",
```

Comment:
This shouldn't be pinned this tightly. Maybe cap at <4; leaving uncapped is fine too. Also we should go ahead and make pytest>=6 while we are at it.

Reply:
Thank you for linking the blog! Amazing read! Never thought about capping so much :)